base/reducedim.jl

```diff
@@ -353,9 +353,16 @@ reduce(op, A::AbstractArray; kw...) = mapreduce(identity, op, A; kw...)
 
 ##### Specific reduction functions #####
 """
-    sum(A::AbstractArray; dims)
+    sum(A::AbstractArray; [dims], [weights::AbstractArray])
 
-Sum elements of an array over the given dimensions.
+Compute the sum of array `A`.
+If `dims` is provided, return an array of sums over these dimensions.
+If `weights` is provided, return the weighted sum(s). `weights` must be
+either an array of the same size as `A` if `dims` is omitted,
+or a vector with the same length as `size(A, dims)` if `dims` is provided.
```

Comment:
Why should `weights` have to be an array, as opposed to just an iterable?

Reply:
That's how things work currently in StatsBase. I guess we could make this more general at least for the simple case where `dims=:`. But that would require adding a separate method as `@simd` only works when using indices.